openwisp_controller/config/whois/service.py

```diff
@@ -108,11 +108,21 @@ def trigger_whois_lookup(self):
         Trigger WHOIS lookup based on the conditions of `_need_whois_lookup`.
         Task is triggered on commit to ensure redundant data is not created.
         """
-        if self._need_whois_lookup(self.device.last_ip):
+        new_ip = self.device.last_ip
+        if self._need_whois_lookup(new_ip):
             transaction.on_commit(
                 lambda: fetch_whois_details.delay(
                     device_pk=self.device.pk,
                     initial_ip_address=self.device._initial_last_ip,
-                    new_ip_address=self.device.last_ip,
+                    new_ip_address=new_ip,
+                )
+            )
+        # we are only checking for 2 of the above conditions of `_need_whois_lookup`
+        # as we need to ensure that the fuzzy locations are managed only if there is
+        # existing WHOIS record
```

Comment:
This comment looks confusing to me, we are managing fuzzy locations in `fetch_whois_details` task also

Reply:
I have updated the comment for better clarity